### ISSUE TITLE
[a11y]: Even in disabled state, the text content should be visible.

### ISSUE THREAD

**Issue by Vinodbs** [carbon-design-system/carbon#16336]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.50.0

### React version

v17.0.2

### Automated testing tool and ruleset

Manual

### Description

All fields should have visible text lables for easy user understanding

Even in disabled state, the text content should be visible.

### Reproduction/example

https://stackblitz.com/edit/github-lu2ax9?file=src%2FApp.jsx

### Steps to reproduce

Run the code
Observe the color difference for the disabled input field

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by guidari** [comment]
Hey @Vinodbs the Stackblitz is not working for me.

![Screenshot 2024-05-07 at 10 53 07](https://github.com/carbon-design-system/carbon/assets/52183462/cada3979-0f58-4e99-a5f6-849f7019cf1f)

**Comment by alisonjoseph** [comment]
@Vinodbs Looks like your code sample is broken, however something to note is that disabled states are not subject to color contrast a11y rules. https://carbondesignsystem.com/elements/color/overview/#disabled

This might be what you were trying to show?
```js
<TextInput
  placeholder="placeholder text"
  labelText="Text input label"
  disabled
/>
```

**Comment by thyhmdo** [comment]
We purposely have the disabled state with low-contrast text. If you need this text to be readable, we recommend using the read-only state.
